node/src/actors/chain_manager/mod.rs

```diff
@@ -1488,6 +1498,37 @@ impl ChainManager {
             Some(sync_target.superblock.checkpoint * superblock_period)
         }
     }
+
+    /// Let JSON-RPC clients know that the blocks in the previous superblock can now
+    /// be considered consolidated
+    fn notify_previous_superblock_consolidation(&mut self, ctx: &mut Context<ChainManager>) {
+        let superblock_period = u32::from(self.consensus_constants().superblock_period);
+        let final_epoch = self
+            .chain_state_snapshot
+            .highest_persisted_superblock
+            .checked_mul(superblock_period)
+            .expect("Multiplying a superblock ID by `superblock_period` should never overflow");
+        let initial_epoch = final_epoch.saturating_sub(superblock_period);
+        let beacons = self.handle(
+            GetBlocksEpochRange::new_with_limit(initial_epoch..final_epoch, 0),
+            ctx,
+        );
+
+        // Get superblock from chain state snapshot
+        let superblock = self.chain_state.superblock_state.get_current_superblock();
```

Comment:
This is the current superblock, so this function will only be valid if called after the superblock votes are counted, but before the next superblock is created. To avoid future problems, I would prefer to pass `superblock` as a parameter to `notify_previous_superblock_consolidation`, and use the id of that superblock to calculate the block epoch range. That way this function can be called at any point in time, even if the superblock has been consolidated long ago.